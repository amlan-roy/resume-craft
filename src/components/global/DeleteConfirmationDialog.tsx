import {
  AlertDialog,
  AlertDialogContent,
  AlertDialogTitle,
  AlertDialogDescription,
  AlertDialogCancel,
  AlertDialogAction,
} from "@/components/ui/alert-dialog";
import React from "react";
import {
  AlertDialogHeader,
  AlertDialogFooter,
} from "@/components/ui/alert-dialog";
import { AlertDialogProps } from "@radix-ui/react-alert-dialog";

type DeleteConfirmationDialogProps = {
  onCancel?: () => void;
  onConfirm?: () => void;
  cancelText?: string;
  confirmText?: string;
  description?: string;
  open?: boolean;
  title?: string;
} & AlertDialogProps;

const DeleteConfirmationDialog: React.FC<DeleteConfirmationDialogProps> = ({
  onCancel,
  onConfirm,
  cancelText = "Cancel",
  confirmText = "Continue",
  description = "This action cannot be undone. This will permanently the data that you have entered for this section.",
  title = "Do you want to delete this section?",
  open,
  ...rest
}) => {
  return (
    <AlertDialog open={open} {...rest}>
      <AlertDialogContent>
        <AlertDialogHeader>
          <AlertDialogTitle>{title}</AlertDialogTitle>
          <AlertDialogDescription>{description}</AlertDialogDescription>
        </AlertDialogHeader>
        <AlertDialogFooter>
          <AlertDialogCancel
            onClick={() => {
              onCancel?.();
            }}
            type="button"
          >
            {cancelText}
          </AlertDialogCancel>
          <AlertDialogAction
            onClick={() => {
              onConfirm?.();
            }}
            type="button"
          >
            {confirmText}
          </AlertDialogAction>
        </AlertDialogFooter>
      </AlertDialogContent>
    </AlertDialog>
  );
};
export default DeleteConfirmationDialog;
