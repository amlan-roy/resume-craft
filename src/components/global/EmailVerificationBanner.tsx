import React from "react";
import { Button } from "@/components/ui/button";

type EmailVerificationBannerProps = {
  onDismiss?: Function;
};

const EmailVerificationBanner: React.FC<EmailVerificationBannerProps> = ({
  onDismiss,
}) => {
  return (
    <div className="bg-brand-neutral-5 text-brand-neutral-9 p-4 flex flex-row items-center">
      <p className="text-center w-full text-wrap">
        Your email is not verified. Please verify your email to access all
        features. Check your inbox for verification email.
      </p>
      {onDismiss && (
        <Button variant={"ghost"} onClick={() => onDismiss?.()}>
          X
        </Button>
      )}
    </div>
  );
};
export default EmailVerificationBanner;
