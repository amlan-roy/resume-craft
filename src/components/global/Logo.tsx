import Image from "next/image";
import React from "react";
import Link from "next/link";
import { cn } from "@/lib/utils";

type LogoProps = {
  linkify?: boolean;
  hideText?: boolean;
  className?: string;
  imageClassName?: string;
  textClassName?: string;
};

const LogoContainer: React.FC<{
  linkify?: boolean;
  children?: React.ReactNode;
  className?: string;
}> = ({ linkify, children, className }) => {
  return linkify ? (
    <Link
      data-testid="global__logo-linkified"
      href={"/"}
      className={cn(["flex justify-center items-center", className])}
    >
      {children}
    </Link>
  ) : (
    <div
      data-testid="global__logo"
      className={cn(["flex justify-center items-center", className])}
    >
      {children}
    </div>
  );
};

const Logo: React.FC<LogoProps> = ({
  linkify,
  hideText,
  className,
  imageClassName,
  textClassName,
}) => {
  return (
    <LogoContainer linkify={linkify} className={className}>
      <Image
        src={"/icon.png"}
        alt={"Resume Craft Icon"}
        width={36}
        height={36}
        className={cn(["w-9 h-9 sm:w-full sm:h-full", imageClassName])}
      />
      {!hideText && (
        <div
          className={cn([
            "flex flex-col text-brand-neutral-12 mx-3 justify-center items-center",
            textClassName,
          ])}
        >
          <p className="text-base sm:text-xl font-extrabold leading-none sm:leading-none">
            RESUME
          </p>
          <p className="text-base sm:text-xl leading-none sm:leading-none ">
            <span className="font-light"> - </span>craft
            <span className="font-light"> - </span>
          </p>
        </div>
      )}
    </LogoContainer>
  );
};
export default Logo;
