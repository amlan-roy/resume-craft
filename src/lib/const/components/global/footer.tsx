import { FaGithub, FaLinkedin, FaTwitter, FaGlobeAsia } from "react-icons/fa";

export const LINKS = [
  {
    url: "https://www.amlan-roy.com/",
    description: "Amlan Roy's Portfolio Website's Link",
    icon: <FaGlobeAsia size={32} className="w-4 h-4 md:w-6 md:h-6" />,
  },
  {
    url: "https://github.com/amlan-roy/",
    description: "Amlan Roy' GitHub Link",
    icon: <FaGithub size={32} className="w-4 h-4 md:w-6 md:h-6" />,
  },
  {
    url: "https://www.linkedin.com/in/amlan-roy/",
    description: "Amlan Roy's LinkedIn Link",
    icon: <FaLinkedin size={32} className="w-4 h-4 md:w-6 md:h-6" />,
  },
  {
    url: "https://twitter.com/_royamlan_",
    description: "Amlan Roy's Twitter Link",
    icon: <FaTwitter size={32} className="w-4 h-4 md:w-6 md:h-6" />,
  },
];
