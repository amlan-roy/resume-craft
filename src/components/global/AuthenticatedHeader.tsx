"use client";

import React, { useEffect, useState } from "react";
import { sendEmailVerification } from "firebase/auth";
import { CircleUserRoundIcon } from "lucide-react";
import { useTheme } from "next-themes";
import useLocalStorage from "@/lib/hooks/useLocalStorage";
import { cn } from "@/lib/utils";
import { auth } from "@/lib/utils/firebase/config";
import {
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuSeparator,
  DropdownMenuTrigger,
} from "@/components/ui/dropdown-menu";
import { Label } from "@/components/ui/label";
import { Switch } from "@/components/ui/switch";
import LogoutButton from "@/components/auth/LogoutButton";
import Logo from "@/components/global/Logo";
import { Button } from "../ui/button";
import { useToast } from "../ui/use-toast";
import EmailVerificationBanner from "./EmailVerificationBanner";

const ThemeSwitch = ({ className }: { className?: string }) => {
  const { setTheme, theme } = useTheme();
  return (
    <div className={cn(["flex gap-6 items-center", className])}>
      <Switch
        id="dark-mode"
        checked={theme === "dark"}
        onCheckedChange={() => {
          theme === "dark" ? setTheme("light") : setTheme("dark");
        }}
      />
      <Label htmlFor="dark-mode">Dark Mode</Label>
    </div>
  );
};

type AuthenticatedHeaderProps = {
  hideLogout?: boolean;
};

const AuthenticatedHeader: React.FC<AuthenticatedHeaderProps> = ({
  hideLogout,
}) => {
  const [emailVerified, setEmailVerified] = React.useState(
    auth.currentUser?.emailVerified
  );

  const [bannerDismissed, setBannerDismissed] = useState(false);

  useEffect(() => {
    setEmailVerified(auth.currentUser?.emailVerified);
  }, [auth.currentUser]);

  const { toast: displayToast } = useToast();

  const [lastAuthRequestSent, setLastAuthRequestSent] = useLocalStorage(
    `${auth.currentUser?.uid}-last-auth-request-sent`
  );

  const resendVerificationEmail = async () => {
    try {
      if (
        lastAuthRequestSent &&
        Date.now() - parseInt(lastAuthRequestSent) < 60000
      ) {
        displayToast({
          title: "Wait a minute!",
          description: "Please wait before sending another email.",
          variant: "destructive",
        });
        return;
      }
      if (auth.currentUser && !auth.currentUser?.emailVerified) {
        await sendEmailVerification(auth.currentUser);
        setLastAuthRequestSent(Date.now().toString());
        displayToast({
          title: "Email sent!",
          description: "Please check your inbox for the verification email.",
          variant: "default",
        });
      }
    } catch (error) {
      displayToast({
        title: "An error occurred",
        description: "Please try again later.",
        variant: "destructive",
      });
    }
  };

  return (
    <>
      {!emailVerified && !bannerDismissed && (
        <EmailVerificationBanner
          onDismiss={() => {
            setBannerDismissed(true);
          }}
        />
      )}
      <header
        data-testid="global__header"
        className="p-4 w-full flex justify-between mx-auto max-w-screen-xl"
      >
        <Logo linkify />
        <div className="flex items-center">
          {!hideLogout ? (
            <DropdownMenu>
              <DropdownMenuTrigger title="See More" className="p-4">
                <CircleUserRoundIcon />
              </DropdownMenuTrigger>
              <DropdownMenuContent className="mr-4 p-4">
                <div className="text-brand-neutral-9 text-base font-semibold">
                  Hi {auth.currentUser?.displayName || "there"}!
                </div>
                <DropdownMenuSeparator className="my-4" />
                <ThemeSwitch />
                {emailVerified === false && (
                  <>
                    <DropdownMenuSeparator className="my-4" />
                    <Button
                      variant={"outline"}
                      onClick={resendVerificationEmail}
                      className="w-full"
                    >
                      Resend Verification Email
                    </Button>
                  </>
                )}
                <DropdownMenuSeparator className="my-4" />
                <LogoutButton buttonClass="w-full" />
              </DropdownMenuContent>
            </DropdownMenu>
          ) : (
            <ThemeSwitch />
          )}
        </div>
      </header>
    </>
  );
};
export default AuthenticatedHeader;
