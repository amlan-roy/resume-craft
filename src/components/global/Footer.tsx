import React from "react";
import Logo from "@/components/global/Logo";
import {
  FaHeart,
  FaGithub,
  FaLinkedin,
  FaTwitter,
  FaGlobeAsia,
} from "react-icons/fa";
import Link from "next/link";

type FooterProps = {};

const LINKS = [
  {
    url: "https://amlan-roy.github.io/",
    description: "Amlan Roy's Portfolio Website's Link",
    icon: <FaGlobeAsia size={32} className="w-4 h-4 md:w-6 md:h-6" />,
  },
  {
    url: "https://github.com/amlan-roy/",
    description: "Amlan Roy' GitHub Link",
    icon: <FaGithub size={32} className="w-4 h-4 md:w-6 md:h-6" />,
  },
  {
    url: "https://www.linkedin.com/in/amlan-roy/",
    description: "Amlan Roy's LinkedIn Link",
    icon: <FaLinkedin size={32} className="w-4 h-4 md:w-6 md:h-6" />,
  },
  {
    url: "https://twitter.com/_royamlan_",
    description: "Amlan Roy's Twitter Link",
    icon: <FaTwitter size={32} className="w-4 h-4 md:w-6 md:h-6" />,
  },
];

const Footer: React.FC<FooterProps> = () => {
  return (
    <footer
      data-testid="global__footer"
      className="md:h-20 bg-brand-primary-green-10 dark:bg-brand-primary-green-9 py-5"
    >
      <div className="flex h-8 justify-between w-full max-w-xl mx-auto px-4">
        <Logo
          linkify
          className="hidden sm:flex"
          imageClassName="w-10 h-10 object-contain"
          textClassName="text-brand-neutral-2"
        />
        <div className="text-brand-neutral-2 flex items-center justify-center text-base w-full flex-wrap ml-4">
          <div className="flex items-center md:text-base text-sm flex-nowrap text-nowrap ">
            <p>{"Made with"}</p>
            <FaHeart className="text-red-500 mx-2" />
            <p>
              {"by Amlan Roy"}
              <span className="ml-3">{"|"}</span>
            </p>
          </div>
          <div className="flex flex-nowrap mx-2 gap-2 justify-around items-center">
            {LINKS.map(({ url, description, icon }, index) => (
              <Link
                href={url}
                key={`footer-icons-${index}`}
                aria-label={description}
                className=""
              >
                {icon}
              </Link>
            ))}
          </div>
        </div>
      </div>
    </footer>
  );
};
export default Footer;
