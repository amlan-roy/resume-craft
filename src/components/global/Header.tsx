"use client";

import React from "react";
import { useTheme } from "next-themes";
import { Label } from "@/components/ui/label";
import { Switch } from "@/components/ui/switch";
import Logo from "@/components/global/Logo";

type HeaderProps = {};

const Header: React.FC<HeaderProps> = () => {
  const { setTheme, theme } = useTheme();
  return (
    <header className="p-4 w-full flex justify-between mx-auto max-w-screen-xl">
      <Logo linkify />
      <div className="flex items-center space-x-2 mx-2">
        <Switch
          id="dark-mode"
          checked={theme === "dark"}
          onCheckedChange={() => {
            theme === "dark" ? setTheme("light") : setTheme("dark");
          }}
        />
        <Label htmlFor="dark-mode">Dark Mode</Label>
      </div>
    </header>
  );
};
export default Header;
