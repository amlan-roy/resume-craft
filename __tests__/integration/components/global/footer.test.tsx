import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { LINKS } from "@/lib/const/components/global/footer";
import Footer from "@/components/global/Footer";

describe("integration:components/global/footer", () => {
  it("snapshot", () => {
    const tree = render(<Footer />);
    expect(tree.container).toMatchSnapshot();
  });

  it("renders the footer with all expected elements", () => {
    const tree = render(<Footer />);

    const footer = screen.getByTestId("global__footer");
    const logo = screen.getByTestId("global__logo-linkified");
    const authorText = screen.getByText(/Made with/i);
    const heartIcon = tree.baseElement.querySelector("svg");
    const authorName = screen.getByText(/Amlan Roy/i);
    const socialLinks = screen
      .getAllByRole("link")
      .filter((element) => element.getAttribute("href") !== "/home");

    expect(footer).toBeInTheDocument();
    expect(logo).toBeInTheDocument();
    expect(authorText).toBeInTheDocument();
    expect(heartIcon).toBeInTheDocument();
    expect(authorName).toBeInTheDocument();
    expect(socialLinks).toHaveLength(4);
  });

  it("renders the footer with all expected links", () => {
    render(<Footer />);

    const links = screen
      .getAllByRole("link")
      .filter((element) => element.getAttribute("href") !== "/home");

    links.forEach((linkElement, index) => {
      const expectedUrl = LINKS[index].url;
      expect(linkElement.getAttribute("href")).toStrictEqual(expectedUrl);
    });
  });
});
