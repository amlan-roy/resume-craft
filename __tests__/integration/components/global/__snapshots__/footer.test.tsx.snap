// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration:components/global/footer snapshot 1`] = `
<div>
  <footer
    class="md:h-20 bg-brand-primary-green-10 dark:bg-brand-primary-green-9 py-5 fixed bottom-0 left-0 right-0"
    data-testid="global__footer"
  >
    <div
      class="flex h-8 justify-between w-full max-w-xl mx-auto px-4"
    >
      <a
        class="justify-center items-center hidden sm:flex"
        data-testid="global__logo-linkified"
        href="/"
      >
        <img
          alt="Resume Craft Icon"
          class="sm:w-full sm:h-full w-10 h-10 object-contain"
          data-nimg="1"
          decoding="async"
          height="36"
          loading="lazy"
          src="/_next/image?url=%2Ficon.png&w=96&q=75"
          srcset="/_next/image?url=%2Ficon.png&w=48&q=75 1x, /_next/image?url=%2Ficon.png&w=96&q=75 2x"
          style="color: transparent;"
          width="36"
        />
        <div
          class="flex flex-col mx-3 justify-center items-center text-brand-neutral-2"
        >
          <p
            class="text-base sm:text-xl font-extrabold leading-none sm:leading-none"
          >
            RESUME
          </p>
          <p
            class="text-base sm:text-xl leading-none sm:leading-none "
          >
            <span
              class="font-light"
            >
               - 
            </span>
            craft
            <span
              class="font-light"
            >
               - 
            </span>
          </p>
        </div>
      </a>
      <div
        class="text-brand-neutral-2 flex items-center justify-center text-base w-full flex-wrap ml-4"
      >
        <div
          class="flex items-center md:text-base text-sm flex-nowrap text-nowrap "
        >
          <p>
            Made with
          </p>
          <svg
            class="text-red-500 mx-2"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
            />
          </svg>
          <p>
            by Amlan Roy
            <span
              class="ml-3"
            >
              |
            </span>
          </p>
        </div>
        <div
          class="flex flex-nowrap mx-2 gap-2 justify-around items-center"
        >
          <a
            aria-label="Amlan Roy's Portfolio Website's Link"
            class=""
            href="https://www.amlan-roy.com/"
            target="_blank"
          >
            <svg
              class="w-4 h-4 md:w-6 md:h-6"
              fill="currentColor"
              height="32"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 496 512"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M248 8C111.03 8 0 119.03 0 256s111.03 248 248 248 248-111.03 248-248S384.97 8 248 8zm-11.34 240.23c-2.89 4.82-8.1 7.77-13.72 7.77h-.31c-4.24 0-8.31 1.69-11.31 4.69l-5.66 5.66c-3.12 3.12-3.12 8.19 0 11.31l5.66 5.66c3 3 4.69 7.07 4.69 11.31V304c0 8.84-7.16 16-16 16h-6.11c-6.06 0-11.6-3.42-14.31-8.85l-22.62-45.23c-2.44-4.88-8.95-5.94-12.81-2.08l-19.47 19.46c-3 3-7.07 4.69-11.31 4.69H50.81C49.12 277.55 48 266.92 48 256c0-110.28 89.72-200 200-200 21.51 0 42.2 3.51 61.63 9.82l-50.16 38.53c-5.11 3.41-4.63 11.06.86 13.81l10.83 5.41c5.42 2.71 8.84 8.25 8.84 14.31V216c0 4.42-3.58 8-8 8h-3.06c-3.03 0-5.8-1.71-7.15-4.42-1.56-3.12-5.96-3.29-7.76-.3l-17.37 28.95zM408 358.43c0 4.24-1.69 8.31-4.69 11.31l-9.57 9.57c-3 3-7.07 4.69-11.31 4.69h-15.16c-4.24 0-8.31-1.69-11.31-4.69l-13.01-13.01a26.767 26.767 0 0 0-25.42-7.04l-21.27 5.32c-1.27.32-2.57.48-3.88.48h-10.34c-4.24 0-8.31-1.69-11.31-4.69l-11.91-11.91a8.008 8.008 0 0 1-2.34-5.66v-10.2c0-3.27 1.99-6.21 5.03-7.43l39.34-15.74c1.98-.79 3.86-1.82 5.59-3.05l23.71-16.89a7.978 7.978 0 0 1 4.64-1.48h12.09c3.23 0 6.15 1.94 7.39 4.93l5.35 12.85a4 4 0 0 0 3.69 2.46h3.8c1.78 0 3.35-1.18 3.84-2.88l4.2-14.47c.5-1.71 2.06-2.88 3.84-2.88h6.06c2.21 0 4 1.79 4 4v12.93c0 2.12.84 4.16 2.34 5.66l11.91 11.91c3 3 4.69 7.07 4.69 11.31v24.6z"
              />
            </svg>
          </a>
          <a
            aria-label="Amlan Roy' GitHub Link"
            class=""
            href="https://github.com/amlan-roy/"
            target="_blank"
          >
            <svg
              class="w-4 h-4 md:w-6 md:h-6"
              fill="currentColor"
              height="32"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 496 512"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              />
            </svg>
          </a>
          <a
            aria-label="Amlan Roy's LinkedIn Link"
            class=""
            href="https://www.linkedin.com/in/amlan-roy/"
            target="_blank"
          >
            <svg
              class="w-4 h-4 md:w-6 md:h-6"
              fill="currentColor"
              height="32"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
              />
            </svg>
          </a>
          <a
            aria-label="Amlan Roy's Twitter Link"
            class=""
            href="https://twitter.com/_royamlan_"
            target="_blank"
          >
            <svg
              class="w-4 h-4 md:w-6 md:h-6"
              fill="currentColor"
              height="32"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
